presto-raptor/src/test/java/com/facebook/presto/raptor/storage/TestOrcFileRewriter.java

```diff
@@ -641,13 +642,24 @@ public void testRewriterDropThenAddSameColumns(boolean useOptimizedOrcWriter)
                 new BitSet(5));
         assertEquals(info.getRowCount(), 1);
 
-        ConnectorPageSource source = storageManager.getPageSource(
-                uuid,
-                OptionalInt.empty(),
-                ImmutableList.of(3L, 7L, 8L),
-                ImmutableList.of(createVarcharType(5), createVarcharType(20), INTEGER),
-                 TupleDomain.all(),
-                READER_ATTRIBUTES);
+        ConnectorPageSource source = null;
+        try {
+            source = storageManager.getPageSource(
+                    uuid,
+                    OptionalInt.empty(),
+                    ImmutableList.of(3L, 7L, 8L),
+                    ImmutableList.of(createVarcharType(5), createVarcharType(20), INTEGER),
+                    TupleDomain.all(),
+                    READER_ATTRIBUTES);
+        }
+        catch (PrestoException e) {
+            if (useOptimizedOrcWriter) {
+                fail();
+            }
+            else {
+                return;
```

Comment:
Why do we ignore the test failure if the optimized orc writer is disabled?

Reply:
Same logic as the catch block at the end of the test. This pr creates the stream readers with the type in the constructor vs. being passed via readBlock so it fails on creation.